File: api_app/api_connect/lib/domain/service/product_service.dart
import 'package:api_connect/domain/entity/product.dart';
import 'package:api_connect/domain/service/base_service.dart';
import 'package:http/http.dart' as http;

class ProductService extends BaseService {

  ProductService() : super('/product');
  
  Future<List<Product>> findAll() async {
    http.Response response = await http.get(getUri());
    return Product.fromJsonList(response.body);
  }
}

File: api_app/demo/src/main/java/com/example/demo/domain/service/ProductService.java
package com.example.demo.domain.service;

import com.example.demo.domain.entity.Product;
import com.example.demo.domain.repository.ProductRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ProductService {

    final ProductRepository repository;

    public Product save(Product product) {
        return repository.save(product);
    }

    public List<Product> findAll() {
        return repository.findAll();
    }
}
